src/funding/funding.ts

```diff
@@ -0,0 +1,408 @@
+import { TokenService } from "services/TokenService";
+import { fromWei } from "./../services/EthereumService";
+import { AlertService, ShowButtonsEnum } from "./../services/AlertService";
+import { NumberService } from "./../services/NumberService";
+import { IDaoClaimToken, IDaoTransaction } from "./../entities/DealTokenSwap";
+import { DateService } from "services/DateService";
+import { EventMessageType } from "./../resources/elements/primeDesignSystem/types";
+import { EventConfig } from "./../services/GeneralEvents";
+import { EventAggregator } from "aurelia-event-aggregator";
+import { BigNumber } from "ethers";
+import "./funding.scss";
+import { DealService } from "services/DealService";
+import { DealTokenSwap } from "entities/DealTokenSwap";
+import { EthereumService } from "services/EthereumService";
+import { Router } from "aurelia-router";
+import { Utils } from "services/utils";
+import { autoinject, computedFrom } from "aurelia-framework";
+import { IDAO } from "entities/DealRegistrationTokenSwap";
+import { ITokenFunding } from "entities/TokenFunding";
+import { IPSelectItemConfig } from "resources/elements/primeDesignSystem/pselect/pselect";
+import { observable } from "aurelia-typed-observable-plugin";
+import moment from "moment-timezone";
+import { IAlertModel } from "services/AlertService";
+import { IGridColumn } from "resources/elements/primeDesignSystem/pgrid/pgrid";
+import { tokenGridColumns, transactionColumns, claimTokenGridColumns } from "./funding-grid-columns";
+import { toBigNumberJs } from "services/BigNumberService";
+@autoinject
+export class Funding {
+  private transactionColumns: IGridColumn[] = transactionColumns;
+  private tokenGridColumns: IGridColumn[] = tokenGridColumns;
+  private claimTokenGridColumns: IGridColumn[] = claimTokenGridColumns;
+  private deal: DealTokenSwap;
+  private dealId: string;
+  private depositAmount: BigNumber;
+  private loadingDeposits = false;
+  private refSelectToken: HTMLSelectElement;
+  private seeingMore = false;
+  private walletBalance: BigNumber;
+  @observable
+  private selectedToken: number;
```

Comment:
```suggestion
  private selectedToken: number | string;
```

Because below, the `selectedTokenChanged` methods has for `newVal: number | string`

Reply:
Good catch, thanks @hiaux0 ! I updated the code to add this.